### PR TITLE
Rework the specs to be more rspec3 friendly

### PR DESCRIPTION
**Pull request by purbon**
Including integration with a fake statsd server and being sure there are no colliding ports for the connections. More test should be added in a feature.

### CODE REVIEW

**Comment by purbon** [comment]
Hi @jsvd, this is another internal cleanup/improvement on test ... would you be so nice to review it. Is an small improvement to avoid collisions when running integration test with this plugin, so test are not broken "randomly" :panda_face:

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  def closed?
+    @terminated == true
+  end
+
+end
+
+module StatdHelpers
+
+  def random_port
```
Comment (jsvd) [inline spec/spec_helper.rb]: this isn't used, can be deleted

File: spec/outputs/statsd_spec.rb [hunk 2]
```diff
@@ -1,91 +1,47 @@
...
 
-  describe "output a very small float" do
-    config <<-CONFIG
-      input {
-        generator {
-          message => "valid"
-          count => 1
-        }
-      }
+      let(:props) do
```
Comment (jsvd) [inline spec/outputs/statsd_spec.rb]: I'd write this variable in full vs abbreviation

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
+require "logstash/devutils/rspec/spec_helper"
+require "socket"
+
+
+class StatsdServer
+
+  attr_reader :received, :port
+
+  def initialize
+      @sync_lock = Mutex.new
```
Comment (jsvd) [inline spec/spec_helper.rb]: -1 indentation level

**Comment by jsvd** [comment]
a few minor comments, mostly cosmetic. Tests run. Just fix those and LGTM

**Pull request closed without merging**